codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -177,6 +180,16 @@ private MethodSpec finalizeServiceConfigurationMethod() {
                          .addCode("interceptors = $T.mergeLists(interceptors, config.option($T.EXECUTION_INTERCEPTORS));\n",
                                   CollectionUtils.class, SdkClientOption.class);
 
+        if (model.getMetadata().isQueryProtocol()) {
+            builder.addStatement("$T<$T> protocolInterceptors = $T.singletonList(new $T())",
```

Comment:
Can we use poet's ParamaterizedTypeName here instead of parameterizing by hand.

Reply:
Like this?
```
TypeName listType = ParameterizedTypeName.get(List.class, ExecutionInterceptor.class);
builder.addStatement("$T protocolInterceptors = $T.singletonList(new $T())",
                                 listType,
                                 Collections.class,
                                 QueryParametersToBodyInterceptor.class);
```

Reply:
Yes, exactly